Yoast/wordpress-seo
# Dupp 21 handle version mismatch

## Context


We want to handle the situation where Free has been updated to the latest version (where the framework has been moved) but Premium has not (still has the the complete workouts framework). 

In this situation, we want to do the following:
* Only want to show the new (Free version) workouts page. This means the Configuration workout will be visible but the premium workouts are not.
* Show a notification to update Premium ([design](https://www.sketch.com/s/843e6f4b-872d-480a-b54a-943575160f5e/a/m1gOdxk/play)).

## Summary


This PR can be summarized in the following changelog entry:

* Handles the mismatch in versions when Free has been updated but Premium has not. 

## Relevant technical choices:

* Adds a Notice_Presenter which introduces a Yoast-styled WordPress admin notification.
* The implementation of the design of the notification has been approved by @uxkai.

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* Checkout this branch and build.
* Checkout the latest `trunk` in Premium and build.
* Navigate to SEO > Workouts.
* Note that in the SEO menu Workouts is _below_  Redirects.
* Check that `Workouts` in the menu does NOT have a Premium badge.
* There should be a notification that matches the design, and the free versions of the workouts should be shown (with the upsells). 
* Check that the link to "update premium to the latest version" directs you to the correct page.
* In `src/integrations/admin/workouts-integration.php` in `should_update_premium` (line 307) change `17.7` to `17.5` or an older version.
* Refresh the Workouts page.
* Note that in the SEO menu Workouts is now again above Redirects. 
* Check that `Workouts` in the menu does NOT have a Premium badge.
* See the notification has disappeared and the Premium versions of the workouts are shown. 


### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

* Activate an older version of Yoast SEO Premium (not the RC).
* Navigate to SEO > Workouts. 
* Note that in the SEO menu Workouts is _below_  Redirects.
* Check that `Workouts` in the menu does NOT have a Premium badge.
* There should be a notification that matches the design, the link should lead you to update premium. 
* The Free version of the workouts page should be shown, with the configuration workout and the upsell version of the  premium workout cards. 
* Activate the Yoast SEO Premium RC (17.7).
* Navigate to SEO > Workouts.
* Note that in the SEO menu Workouts is now again above Redirects.
* Check that `Workouts` in the menu does NOT have a Premium badge.
* See the notification has disappeared and the Premium versions of the workouts are shown. 

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Other environments


## Documentation


## Quality assurance


Fixes [[DUPP-21](https://yoast.atlassian.net/browse/DUPP-21)]
